comicapi/genericmetadata.py

```diff
@@ -204,96 +252,186 @@ def get_clean_metadata(self, *attributes: str) -> GenericMetadata:
+
+        return [parse_url(url) for url in new_set]
+
+    def credit_dedupe(self, cur: list[Credit], new: list[Credit]) -> list[Credit]:
+        if len(new) == 0:
+            return cur
+        if len(cur) == 0:
+            return new
+
+        # Create dict for deduplication
+        new_dict: dict[str, Credit] = {f"{norm_fold(n['person'])}_{n['role'].casefold()}": n for n in new}
+        cur_dict: dict[str, Credit] = {f"{norm_fold(c['person'])}_{c['role'].casefold()}": c for c in cur}
+
+        # Any duplicates use the 'new' value
+        cur_dict.update(new_dict)
+        return list(cur_dict.values())
+
+    def assign_dedupe(self, new: list[str] | set[str], cur: list[str] | set[str]) -> list[str] | set[str]:
+        """Dedupes normalised (NFKD), casefolded values using 'new' values on collisions"""
+        if len(new) == 0:
+            return cur
+        if len(cur) == 0:
+            return new
+
+        # Create dict values for deduplication
+        new_dict: dict[str, str] = {norm_fold(n): n for n in new}
+        cur_dict: dict[str, str] = {norm_fold(c): c for c in cur}
+
+        if isinstance(cur, list):
+            cur_dict.update(new_dict)
+            return list(cur_dict.values())
+
+        if isinstance(cur, set):
+            cur_dict.update(new_dict)
+            return set(cur_dict.values())
+
+        # All else fails
+        return cur
+
+    def assign_overlay(self, cur: Any, new: Any) -> Any:
+        """Overlay - When the 'new' object has non-None values, overwrite 'cur'(rent) with 'new'."""
+        if new is None:
+            return cur
+        # An empty string can be used to clear a list or set via the commandline
+        if isinstance(new, str) and len(new) == 0:
+            if isinstance(cur, (list, set)):
+                cur.clear()
+                return cur
+            else:
+                return None
+        elif isinstance(new, (list, set)) and len(new) == 0:
+            return cur
+        else:
+            return new
+
+    def assign_add_missing(self, cur: Any, new: Any) -> Any:
+        """Add Missing - Any 'cur(rent)' values that are None or an empty str/list/set add 'new' non-None values"""
+        if new is None:
+            return cur
+        if isinstance(cur, (list, set)) and len(cur) == 0:
+            return new
+        elif isinstance(cur, str) and len(cur) == 0:
+            return new
+        elif cur is None:
+            return new
+        else:
+            return cur
+
+    def assign_combine(self, cur: Any, new: Any) -> Any:
+        """Combine - Combine lists and sets. All non-None values from new replace cur(rent)"""
+        if new is None:
+            return cur
+        if isinstance(new, (list, set)) and isinstance(cur, (list, set)):
+            if len(new) > 0 and isinstance(new, list) and isinstance(new[0], Url):
+                return self.weblinks_dedupe(new, cur)  # type: ignore[arg-type]
+            return self.assign_dedupe(new, cur)
```

Comment:
This fails when there are no new links and there is a current link. Probably a good test case.